packages/adapter/src/request-building/authentication.ts

```diff
@@ -74,32 +73,16 @@ export function buildParameters(options: CachedBuildRequestOptions): Authenticat
     cookies: {},
   };
 
-  const { securitySchemeSecrets } = options;
-  // Security schemes originate from 'config.json' and contain all of the authentication details
-  // but the actual secrets and values needed by the Airnode deployment will come from 'secrets.env'
-  if (!securitySchemeSecrets || isEmpty(securitySchemeSecrets)) {
+  const { credentials } = options;
+  if (!credentials) {
     return initialParameters;
   }
 
   // API security schemes also originate from 'config.json' and specify which schemes should be used
-  const apiSecuritySchemes = options.ois.apiSpecifications.components.securitySchemes;
-  if (isEmpty(apiSecuritySchemes)) {
+  const apiSecurityScheme = options.ois.apiSpecifications.components.securitySchemes[credentials.securityScheme];
+  if (!apiSecurityScheme) {
     return initialParameters;
   }
 
-  const apiSecuritySchemeNames = Object.keys(apiSecuritySchemes);
-
-  return apiSecuritySchemeNames.reduce((authentication, apiSecuritySchemeName) => {
```

Comment:
Btw. there are some examples in the code base (adapter and deployer pckg) where the `securitySchemes` field is an array not an object. Can you fix that? 

We could also create an issue or brainstorm how to keep the examples and docs up to date...